frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditCheckbox.tsx

```diff
@@ -247,10 +247,8 @@ export const EditCheckbox = ({ field }: EditCheckboxProps): JSX.Element => {
                   precision={0}
                   flex={1}
                   showSteppers={false}
-                  onChange={(val) => {
-                    // Only allow numeric inputs and return it as a number
-                    const numericValue = parseInt(val.replace(/\D/g, ''))
-                    onChange(isNaN(numericValue) ? val : numericValue)
+                  onChange={(_, valNum) => {
+                    onChange(isNaN(valNum) ? '' : valNum)
```

Comment:
i think we should still not trigger onChange if value is not valid:

```tsx
 onChange={(_, valNum) => {
  if (!isNaN(valNum)) return
  return onChange(valNum)
```

Reply:
currently it's still only possible to enter digits 0-9, so the only way it will be NaN is if the input value is `''`. is there any particular reason to prefer either?

Reply:
we won't suddenly clear the input if an invalid character is entered, and instead just do nothing.